src/network/cni/exec.rs

```diff
@@ -0,0 +1,144 @@
+//! CNI plugin interaction via command execution
+
+use anyhow::{bail, Context, Result};
+use derive_builder::Builder;
+use getset::Getters;
+use std::{
+    collections::HashMap,
+    io::{Read, Write},
+    path::Path,
+    process::{Command, Stdio},
```

Comment:
Still haven't seen the full PR, but should be we using tokio process here instead?

Reply:
Yes, this makes handling a bit more complicated because async functions have no drop. But I implemented it this way and now it seams to work seamlessly. :+1: